packages/python/plotly/_plotly_utils/utils.py

```diff
@@ -95,6 +96,14 @@ def default(self, obj):
         Therefore, we only anticipate either unknown iterables or values here.
 
         """
+        numpy = get_module("numpy", should_load=False)
+        if numpy:
+            if isinstance(obj, numpy.ndarray) and numpy.issubdtype(
+                obj.dtype, numpy.number
+            ):
+                if numpy.all(numpy.isfinite(obj)):
+                    self.unsafe = True
```

Comment:
Not sure if I am missing something. We can test all the numpy arrays that are in the dict being encoded, but we'd not catch nans and infs in any sublists, right?

Reply:
Yes this will work only for numpy arrays but for other kinds of objects the existing process (encoding - decoding - reencoding) will still be executed. Or at least this is the intended behaviour :-), maybe I am the one missing something! I need to write tests anyway.

Reply:
>  but for other kinds of objects the existing process (encoding - decoding - reencoding) will still be executed

I don't think this is the case, since as I understand this class, `encode()` is not called recursively, or is it?